core/src/main/java/org/hibernate/ogm/persister/impl/OgmEntityPersister.java

```diff
@@ -526,6 +543,47 @@ private TupleContext createTupleContext() {
 		);
 	}
 
+
+	public void createIndex(OgmIndexSpec indexSpec)
+	{
+		gridDialect.createIndex(indexSpec);
+	}
+
+	public List<OgmIndexSpec> getIndexSpec()
+	{
+		return gridDialect.getIndexSpec(tableName,indexAnnotations);
+	}
+
+	private Map<String, Annotation> getOgmIndexAnnotations(Class cls) {
```

Comment:
The persister is not the right place to deal with annotation retrieval. It works on meta-data retrieved and validated in previous bootstrap phases.

Assuming we go the path I suggested and re-use `@Table#indexes()`, schema definers would be able to retrieve the configured indexes from the passed `Database` objects and its table mappings. That way also XML-based mappings will be applied and it's abstracted wether fields/or getters have been annotated.

For store-specific settings (should we need them), there is `OptionsService` which exposes these settings (again in a unified manner, so it's the same handling for annotation and API configuration).